pkg/controllers/switchedcontroller/switched_controller.go

```diff
@@ -0,0 +1,127 @@
+package switchedcontroller
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+type ControllerWithSwitch struct {
+	delegateName       string
+	delegateFactoryFn  DelegateFactoryFunc
+	delegateController factory.Controller
```

Comment:
Do we actually need to store a reference to the delegate controller here? Can we just blindly create a new delegate controller and run it in a goroutine with the right context and when it gets shutdown it automatically get cleaned up?